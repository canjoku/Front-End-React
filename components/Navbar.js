import { Component } from 'react'
import Link from 'next/link'

class Navbar extends Component {
  render () {
    return (
      <nav className="nav">
        <input type="checkbox" class="nav__cb" id="menu-cb"/>
        <div className="nav__content">
          <ul className="nav__items">
            <li className="nav__item">
              <span className="nav__item-text">
                <Link href='/index'><a>Home</a></Link>
              </span>
            </li>
            <li className="nav__item">
              <span className="nav__item-text">
              <Link href='/about'><a>About</a></Link>
              </span>
            </li>
            <li className="nav__item">
              <span className="nav__item-text">
                <Link href='/projects'><a>Projects</a></Link>
              </span>
            </li>
            <li className="nav__item">
              <span className="nav__item-text">
                <Link href='/articles'><a>Articles</a></Link>
              </span>
            </li>
          </ul>
        </div>
        <label className="nav__btn" for="menu-cb"></label>
      </nav>
    )
  }
}

export default Navbar




