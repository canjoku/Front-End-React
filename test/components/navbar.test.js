import Navbar from '../../components/Navbar'

test('<Navbar /> renders correctly', () => {
  const component = shallow(<Navbar />)
  expect(component).toMatchSnapshot()
})
