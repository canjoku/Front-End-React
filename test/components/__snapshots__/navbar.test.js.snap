// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="c-navbar__checkbox"
          id="nav-toggle"
          type="checkbox"
        />,
        <label
          className="c-navbar__button"
          for="nav-toggle"
        >
          <span
            className="c-navbar__icon"
          />
        </label>,
        <div
          className="c-navbar__background"
        />,
        <nav
          className="c-navbar__nav"
        >
          <ul
            className="c-navbar__list"
          >
            <li
              className="c-navbar__item"
            >
              <Link
                href="/index"
              >
                <a
                  className="c-navbar__link"
                >
                  Home
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/about"
              >
                <a
                  className="c-navbar__link"
                >
                  About
                  <a />
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/projects"
              >
                <a
                  className="c-navbar__link"
                >
                  Projects
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/articles"
              >
                <a
                  className="c-navbar__link"
                >
                  Articles
                </a>
              </Link>
            </li>
          </ul>
        </nav>,
      ],
      "className": "c-navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-navbar__checkbox",
          "id": "nav-toggle",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="c-navbar__icon"
          />,
          "className": "c-navbar__button",
          "for": "nav-toggle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-navbar__icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-navbar__background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="c-navbar__list"
          >
            <li
              className="c-navbar__item"
            >
              <Link
                href="/index"
              >
                <a
                  className="c-navbar__link"
                >
                  Home
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/about"
              >
                <a
                  className="c-navbar__link"
                >
                  About
                  <a />
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/projects"
              >
                <a
                  className="c-navbar__link"
                >
                  Projects
                </a>
              </Link>
            </li>
            <li
              className="c-navbar__item"
            >
              <Link
                href="/articles"
              >
                <a
                  className="c-navbar__link"
                >
                  Articles
                </a>
              </Link>
            </li>
          </ul>,
          "className": "c-navbar__nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/index"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Home
                  </a>
                </Link>
              </li>,
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/about"
                >
                  <a
                    className="c-navbar__link"
                  >
                    About
                    <a />
                  </a>
                </Link>
              </li>,
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/projects"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Projects
                  </a>
                </Link>
              </li>,
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/articles"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Articles
                  </a>
                </Link>
              </li>,
            ],
            "className": "c-navbar__list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  href="/index"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Home
                  </a>
                </Link>,
                "className": "c-navbar__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    className="c-navbar__link"
                  >
                    Home
                  </a>,
                  "href": "/index",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "className": "c-navbar__link",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  href="/about"
                >
                  <a
                    className="c-navbar__link"
                  >
                    About
                    <a />
                  </a>
                </Link>,
                "className": "c-navbar__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    className="c-navbar__link"
                  >
                    About
                    <a />
                  </a>,
                  "href": "/about",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "About",
                      <a />,
                    ],
                    "className": "c-navbar__link",
                  },
                  "ref": null,
                  "rendered": Array [
                    "About",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                  ],
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  href="/projects"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Projects
                  </a>
                </Link>,
                "className": "c-navbar__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    className="c-navbar__link"
                  >
                    Projects
                  </a>,
                  "href": "/projects",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Projects",
                    "className": "c-navbar__link",
                  },
                  "ref": null,
                  "rendered": "Projects",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  href="/articles"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Articles
                  </a>
                </Link>,
                "className": "c-navbar__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <a
                    className="c-navbar__link"
                  >
                    Articles
                  </a>,
                  "href": "/articles",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Articles",
                    "className": "c-navbar__link",
                  },
                  "ref": null,
                  "rendered": "Articles",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="c-navbar__checkbox"
            id="nav-toggle"
            type="checkbox"
          />,
          <label
            className="c-navbar__button"
            for="nav-toggle"
          >
            <span
              className="c-navbar__icon"
            />
          </label>,
          <div
            className="c-navbar__background"
          />,
          <nav
            className="c-navbar__nav"
          >
            <ul
              className="c-navbar__list"
            >
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/index"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/about"
                >
                  <a
                    className="c-navbar__link"
                  >
                    About
                    <a />
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/projects"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Projects
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/articles"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Articles
                  </a>
                </Link>
              </li>
            </ul>
          </nav>,
        ],
        "className": "c-navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-navbar__checkbox",
            "id": "nav-toggle",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="c-navbar__icon"
            />,
            "className": "c-navbar__button",
            "for": "nav-toggle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-navbar__icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-navbar__background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="c-navbar__list"
            >
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/index"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/about"
                >
                  <a
                    className="c-navbar__link"
                  >
                    About
                    <a />
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/projects"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Projects
                  </a>
                </Link>
              </li>
              <li
                className="c-navbar__item"
              >
                <Link
                  href="/articles"
                >
                  <a
                    className="c-navbar__link"
                  >
                    Articles
                  </a>
                </Link>
              </li>
            </ul>,
            "className": "c-navbar__nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="c-navbar__item"
                >
                  <Link
                    href="/index"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Home
                    </a>
                  </Link>
                </li>,
                <li
                  className="c-navbar__item"
                >
                  <Link
                    href="/about"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      About
                      <a />
                    </a>
                  </Link>
                </li>,
                <li
                  className="c-navbar__item"
                >
                  <Link
                    href="/projects"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Projects
                    </a>
                  </Link>
                </li>,
                <li
                  className="c-navbar__item"
                >
                  <Link
                    href="/articles"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Articles
                    </a>
                  </Link>
                </li>,
              ],
              "className": "c-navbar__list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/index"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Home
                    </a>
                  </Link>,
                  "className": "c-navbar__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="c-navbar__link"
                    >
                      Home
                    </a>,
                    "href": "/index",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Home",
                      "className": "c-navbar__link",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/about"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      About
                      <a />
                    </a>
                  </Link>,
                  "className": "c-navbar__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="c-navbar__link"
                    >
                      About
                      <a />
                    </a>,
                    "href": "/about",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "About",
                        <a />,
                      ],
                      "className": "c-navbar__link",
                    },
                    "ref": null,
                    "rendered": Array [
                      "About",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "a",
                      },
                    ],
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/projects"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Projects
                    </a>
                  </Link>,
                  "className": "c-navbar__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="c-navbar__link"
                    >
                      Projects
                    </a>,
                    "href": "/projects",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Projects",
                      "className": "c-navbar__link",
                    },
                    "ref": null,
                    "rendered": "Projects",
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/articles"
                  >
                    <a
                      className="c-navbar__link"
                    >
                      Articles
                    </a>
                  </Link>,
                  "className": "c-navbar__item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="c-navbar__link"
                    >
                      Articles
                    </a>,
                    "href": "/articles",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Articles",
                      "className": "c-navbar__link",
                    },
                    "ref": null,
                    "rendered": "Articles",
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
